Devtailor/Service-Module
# 41 Change service status (active/inactive/draft)

- https://github.com/buerokratt/Service-Module/issues/41

Also as a part of this PR:
- Ports are updated to run together with training module.
- Added mock folder to support training module logic.

This PR goes together with:
- https://github.com/Devtailor/Training-Module/pull/1 
- https://github.com/buerokratt/DataMapper/pull/7 - not strictly required as datamapper is not used currently, but needed for the port 3005 to work, otherwise it will just be inaccessible. 


Prerequisites to test changing service state:
- make sure training module and datamapper are both running using images with changes required by this PR
- Create a new service using `/services/add` name it for example `car_wash`
- Create a file named `service_car_wash_nlu.tmp` under `DSL/mock1/data/nlu/removed/service_car_wash_nlu.tmp`. Training Module treats files under `removed` as soft deleted/inactive.
- File content should be something like 
```
version: "3.0"
nlu:
  - intent: service_car_wash
    examples: ""
```
- Response in `domain.yml` is optional as soft delete/activate flow does not alter it.
```
responses:
  utter_service_car_wash:
    - text: "#service_car_wash"
```
Now you should have everything to change it's state back and forth.

Sample request to change the state of the service:
```
POST http://localhost:8085/services/status

body:
{
  "id": 1, // id of the service from the database
  "state": "active", // possible values are (active, inactive)
  "type": "POST" // possible values are (POST, GET)
}
```

Sometimes timeout error occurs when both ruuters have been running for a while: https://stackoverflow.com/questions/67792501/facing-io-netty-handler-timeout-readtimeoutexception-null-while-consuming-serve
